src/transformers/models/bark/modeling_bark.py

```diff
@@ -0,0 +1,1429 @@
+                    attention_mask,
+                    head_mask[i],
+                )
+            else:
+                outputs = block(
+                    hidden_states,
+                    past_kv=past_layer_kv,
+                    attention_mask=attention_mask,
+                    head_mask=head_mask[i],
+                    use_cache=use_cache,
+                    output_attentions=output_attentions,
+                )
+
+            hidden_states = outputs[0]
+
+            if use_cache:
+                present_key_values = present_key_values + (outputs[1],)
+
+            if output_attentions:
+                all_self_attentions = all_self_attentions + (outputs[2 if use_cache else 1],)
+
+        hidden_states = self.transformer.ln_f(hidden_states)
+
+        hidden_states = hidden_states.view(output_shape)
+
+        # Add last hidden state
+        if output_hidden_states:
+            all_hidden_states = all_hidden_states + (hidden_states,)
+
+        logits = self.lm_head(hidden_states)
+
+        loss = None
+        if labels is not None:
+            raise NotImplementedError("Training is not implemented yet")
+
+        if not return_dict:
+            return tuple(
+                v for v in [None, logits, present_key_values, all_hidden_states, all_self_attentions] if v is not None
+            )
+
+        return CausalLMOutputWithPast(
+            loss=loss,
+            logits=logits,
+            past_key_values=present_key_values,
+            hidden_states=all_hidden_states,
+            attentions=all_self_attentions,
+        )
+
+    @staticmethod
+    def _reorder_cache(
+        past_key_values: Tuple[Tuple[torch.Tensor]], beam_idx: torch.Tensor
+    ) -> Tuple[Tuple[torch.Tensor]]:
+        """
+        This function is used to re-order the `past_key_values` cache if [`~PreTrainedModel.beam_search`] or
+        [`~PreTrainedModel.beam_sample`] is called. This is required to match `past_key_values` with the correct
+        beam_idx at every generation step.
+        """
+        # Necessary for beam_search
+        return tuple(
+            tuple(past_state.index_select(0, beam_idx.to(past_state.device)) for past_state in layer_past)
+            for layer_past in past_key_values
+        )
+
+
+@add_start_docstrings(
+    "Bark sub-module at the core of the semantic sub-model. It shares the same architecture than the coarse model. It is a GPT-2 like autoregressive model with a language modeling head on top.",
+    BARK_MODULE_START_DOCSTRING,
+)
+class BarkSemanticModule(BarkCausalModule):
+    base_model_prefix = "semantic"
+    config_class = BarkSemanticConfig
+
+
+@add_start_docstrings(
+    "Bark sub-module at the core of the coarse acoustics sub-model. It shares the same architecture than the semantic model. It is a GPT-2 like autoregressive model with a language modeling head on top.",
+    BARK_MODULE_START_DOCSTRING,
+)
+class BarkCoarseAcousticsModule(BarkCausalModule):
```

Comment:
No need for the `Acoustics` in the naming here - there's only one `Coarse` model, so we don't need to append an `Acoustics` on the end since there's no ambiguity

```suggestion
class BarkCoarseModel(BarkCausalModel):
```

-> this is then consistent with what you've done for the semantic model (you've gone with `BarkSemanticModel`, rather than `BarkSemanticTextModel`, which is sufficient!)